src/core/testing/test/util.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { h } from '../../../core/renderer/h';
+import * as util from '../util';
+
+class IonTest {
+  render() {
+    return [
+      h('div', 0, 'hi')
+    ];
+  }
+}
+
+class IonThingOne {
+  render() {
+    return [
+      h('div', 0, 'thing')
+    ];
+  }
+}
+
+class IonThingTwo {
+  x = 18;
+  y = 24;
+  sum() { return this.x + this.y; }
+  render() {
+    return [
+      h('div', { c: 'thing' }, [
+        h('div', { c: 'child-thing' }, 'I am the eldest'),
+        h('div', { c: 'child-thing' }, 'Middle child rules'),
+        h('div', { c: 'child-thing' }, 'No youngest does'),
+        h('div', { c: 'child-thing' }, 'say wut now?'),
+        h('div', { c: 'random-sum' }, this.sum())
+      ])
+    ];
+  }
+}
+
+describe('testing utilities', () => {
+  describe('register', () => {
+    it('returns a platform', () => {
+      const plt = util.register();
+      expect(plt).toBeTruthy();
+      expect(typeof plt.defineComponent).toEqual('function');
+      expect(typeof plt.getComponentMeta).toEqual('function');
+      expect(typeof plt.getContextItem).toEqual('function');
+    });
+
+    it('registers the components', () => {
+      const plt = util.register(
+        [{
```

Comment:
I am hoping to not have to specify the `tagNameMeta` in cases where that is defined in the `@Component` decorator in a `.tsx` file, but I still need to look into how that is done.